<?xml version="1.0"?>
<launch>
  <!-- Load perception configuration -->
  <rosparam command="load" file="$(find orca_perception)/config/perception.yaml"/>
  
  <!-- Start length measurement node -->
  <node name="length_measurement_node" pkg="orca_perception" type="length_measurement_node" output="screen">
    <param name="min_distance" value="$(arg min_distance)" if="$(eval arg('min_distance') != 0.0)"/>
    <param name="max_distance" value="$(arg max_distance)" if="$(eval arg('max_distance') != 0.0)"/>
    <param name="cluster_tolerance" value="$(arg cluster_tolerance)" if="$(eval arg('cluster_tolerance') != 0.0)"/>
    <param name="min_cluster_size" value="$(arg min_cluster_size)" if="$(eval arg('min_cluster_size') != 0)"/>
    <param name="max_cluster_size" value="$(arg max_cluster_size)" if="$(eval arg('max_cluster_size') != 0)"/>
  </node>
  
  <!-- Start object detection node -->
  <node name="object_detector" pkg="orca_perception" type="object_detection.py" output="screen">
    <param name="camera_topic" value="$(arg camera_topic)" if="$(eval arg('camera_topic') != '')"/>
    <param name="detection_method" value="$(arg detection_method)" if="$(eval arg('detection_method') != '')"/>
    <param name="min_area" value="$(arg min_area)" if="$(eval arg('min_area') != 0)"/>
    <rosparam param="target_color_lower" subst_value="True">$(arg target_color_lower)</rosparam>
    <rosparam param="target_color_upper" subst_value="True">$(arg target_color_upper)</rosparam>
  </node>
</launch>
